[DOCS-1540] Initial docs for LLM Evaluation Jobs (#1910)
wandb/docs: models/launch.mdx

1. Set up the evaluation programmatically or in the W&B App, defining the benchmarks to run and whether to generate a leaderboard.
1. Run the evaluation.
1. View and analyze the results and leaderboard.

Each time you run an evaluation with the same destination project, the leaderboard is updated automatically.

## Next steps
- Learn more about the [available evaluation benchmarks](/models/launch/evaluations)
- [Evaluate a model checkpoint](/models/launch/evaluate-model-checkpoint)
- [Evaluate an API-hosted model](/models/launch/evaluate-hosted-model)

## More details

### Pricing
LLM Evaluation Jobs is in **Preview**. During the preview period, there is no charge for LLM Evaluation Jobs.

### Requirements
- To evaluate an OpenAI-compatible API-hosted model:
  - The model must be accessible at a public URL.
  - The API key for the model must be available as a team secret. An organization admin or team admin can create a team secret.
- To evaluate model checkpoint:
  - The model weights must be packaged as a VLLM-compatible artifact.

### Limitations
- The maximum size for a model to evaluate is 86 GB, including context.

> - Each job is limited to two GPUs.

Reviewer: I think we should specify that these limitations are for the Evaluate model checkpoint job

---
description: Evaluate model checkpoints or hosted API models within W&B and analyze the results using automatically generated leaderboards.
title: LLM Evaluation Jobs
---
import PreviewLink from '/snippets/en/_includes/llm-eval-jobs/preview.mdx';

> [W&B LLM Evaluation Jobs](/models/launch) is a framework for measuring and analyzing how an LLM model performs, using infrastructure managed by CoreWeave. Run a comprehensive suite of modern, industry-standard [model evaluation benchmarks](/models/launch/evaluations), then view and analyze and share your model's performance using automatic leaderboards and charts in W&B. LLM Evaluation Jobs removes the complexity of deploying and maintaining advanced compute infrastructure yourself.

Reviewer 2: I feel like "measuring and analyzing how an LLM model performs" is kinda vague. What can we evaluate with this feature?

Author: WDYT about this suggestion?

> [W&B LLM Evaluation Jobs](/models/launch) is a framework for measuring and analyzing how an LLM model performs, using infrastructure managed by CoreWeave. Run a comprehensive suite of modern, industry-standard [model evaluation benchmarks](/models/launch/evaluations), then view and analyze and share your model's performance using automatic leaderboards and charts in W&B. LLM Evaluation Jobs removes the complexity of deploying and maintaining advanced compute infrastructure yourself.

Author: ```suggestion
[W&B LLM Evaluation Jobs](/models/launch) is a benchmarking framework for evaluating an LLM model's performance, using infrastructure managed by CoreWeave. Choose from a comprehensive suite of modern, industry-standard [model evaluation benchmarks](/models/launch/evaluations), then view, analyze, and share the results using automatic leaderboards and charts in W&B. LLM Evaluation Jobs removes the complexity of deploying and maintaining GPU  infrastructure yourself.
```

> [W&B LLM Evaluation Jobs](/models/launch) is a framework for measuring and analyzing how an LLM model performs, using infrastructure managed by CoreWeave. Run a comprehensive suite of modern, industry-standard [model evaluation benchmarks](/models/launch/evaluations), then view and analyze and share your model's performance over time using automatic leaderboards and charts in W&B. LLM Evaluation Jobs removes the complexity of deploying and maintaining advanced compute infrastructure yourself.

Reviewer 3: "over time" seems odd here. I'd drop it, since the evaluation is run at a single model

> <PreviewLink />

Author: ```suggestion
<Note><PreviewLink /></Note>
```

{/*
<CardGroup cols={4}>
<ColabLink url="https://colab.research.google.com/github/wandb/examples/blob/master/colabs/intro/Intro_to_Weights_%26_Biases.ipynb" />
<TryProductLink url="https://wandb.ai/stacey/deep-drive/workspace?workspace=user-lavanyashukla" />
</CardGroup>
*/}

## How it works
Evaluate a model checkpoint or a publicly accessible hosted OpenAI-compatible model in just a few steps:

1. Set up an evaluation job in the W&B App. Define its benchmarks and configuration, such as whether to generate a leaderboard.
1. Launch the evaluation job.
1. View and analyze the results and leaderboard.

> Each time you launch an evaluation job with the same destination project, the leaderboard is updated automatically.

Reviewer 3: nit: leaderboard for that project (not sure if clear)

<Frame>
![Example evalution job leaderboard](/images/models/llm-evaluation-jobs/model-checkpoint-leaderboard-example.png)
</Frame>

## Next steps
- Browse the [Evaluation benchmark catalog](/models/launch/evaluations)
- [Evaluate a model checkpoint](/models/launch/evaluate-model-checkpoint)
- [Evaluate an API-hosted model](/models/launch/evaluate-hosted-model)

## More details

### Pricing

> <PreviewLink />

Author: ```suggestion
LLM Evaluation Jobs evaluates a model checkpoint or hosted API against popular benchmarks on fully-managed CoreWeave compute, with no infrastructure to manage. You pay only for resources consumed, not idle time. Pricing has two components: compute and storage. Compute is free during public preview, and we will announce pricing at general availability. Stored results include metrics and per-example traces saved in W&B Models runs. Storage is billed monthly based on data volume. Every plan includes some free storage. See [Pricing](https://wandb.ai/site/pricing/) for details.
```

> LLM Evaluation Jobs is in **Preview**. During the preview period, there is no charge.

Reviewer 2: ```suggestion
LLM Evaluation Jobs is in **Preview**. There is no charge during the preview period.
```
nit

> LLM Evaluation Jobs is in **Preview**. During the preview period, there is no charge.

Author: ```suggestion
<PreviewLink />
```

### Job limits
An individual evaluation job has these limits:
- The maximum size for a model to evaluate is 86 GB, including context.
- Each job is limited to two GPUs.

### Requirements

> - To evaluate a model checkpoint, the model weights must be packaged as a VLLM-compatible artifact.

Reviewer 3: non-blocking: Can we link to docs for how to prepare a vllm compatible artifact?

> - To evaluate an OpenAI-compatible model, it must be accessible at a public URL, and the API key for authentication must be available as a team secret. An organization admin or team admin can create a team secret.

Reviewer 3: Shoudl we call out the team secret aspect and note you may need a team admin to set them up?

Reviewer 3: Ah I see that below. Personally I like putting these right where its first mentioned, but non-blocking

Author: I put it below because it's not complete, you have to have a model already, and we decided that it was too onerous to give example code for a model. It is more of an "appendix" item, I thought. WDYT?

> - Certain benchmarks use OpenAI for scoring. To run these benchmarks, an OpenAI API key must be available to as a team secret.

Reviewer 3: nit: "use OpenAI models"

> - Certain benchmarks use OpenAI models for scoring. To run these benchmarks, an organization or team admin must configure a team secret with the the API key. This requirement is documented in [Evaluation benchmark catalog](/models/launch/evaluations).

Reviewer 2: ```suggestion
- Certain benchmarks use OpenAI models for scoring. To run these benchmarks, an organization or team admin must configure a team secret with the API key. This requirement is documented in [Evaluation benchmark catalog](/models/launch/evaluations).
```
